File: leetcode-submissions/0605-can-place-flowers/solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0
        flowerbed = [0] + flowerbed + [0]
        for i in range(1, len(flowerbed) - 1, 1):
            if flowerbed[i] == 0 and flowerbed[i-1] == 0 and flowerbed[i+1] == 0:
                count += 1
                flowerbed[i] = 1
            
            if count >= n:
                return True
        return n <= count

File: leetcode-submissions/0605-can-place-flowers/solution.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        //check before and after
        //3, check before, update n, and update flowerbed

        //edge cases...
        if (n == 0) { return true; }
        for (int i = 0; i < flowerbed.length ; i++) {
            if (
                (i == 0 || flowerbed[i-1] == 0) &&
                flowerbed[i] == 0 &&
                (i == flowerbed.length - 1 || flowerbed[i+1] == 0)
            ) {
                flowerbed[i] = 1;
                n--;
                if (n == 0) {
                    return true;
                }
            }
        }
        return false; 
    }
}
